src/metadataGeneration/controllerGenerator.ts

```diff
@@ -51,6 +56,21 @@ export class ControllerGenerator {
       .map(generator => generator.Generate());
   }
 
+  private getMethodSignatureDuplicates(methods: Tsoa.Method[]) {
```

Comment:
Why not add a return type and just return the empty array if there are no results? The method name would suggest getting an array anyway. And then above you could just check for the length of the array.

Reply:
1. Typescript infer return type by itself, and it is more flexible way then explicit definition.
2. It doesnt matter to return an empty array or null, anyway i should add condition, tiny benefit i see if we return empty array, so probably, in future, we may concat it with another array without checks that value may be null, but for now returing an empty array left more questions then real sense.

Reply:
I would disagree.

1. While TS can infer the type, this is about making the code more readable for future readers. I don't want to parse the whole method before I know what's returned. 
2. Sure it matters. If you make it an array, you can directly return `result` instead of having to add the check here, which makes the code more concise. The method name suggests that you return an array, which you sometimes do. So `null` really comes as a surprise when calling this method. It might be different if the method were called `checkMethodSignaturesForDuplicates`.

This is just my personal preference though. What do you think @WoH?

Reply:
1. If you use modern IDE, it  can help you to check method signature. On other hand if you prefer to define return type explicit, it assumes that you do it in whole project, as result if you decide to change return type you should manually change all the types, it is not a big deal, but if typescript infer return type, most of types will be updated automatic that is more handful.
2. Ok, it has sense. I'll fix.